### PR TITLE
Add Schema.find_column_name method

### PR DESCRIPTION
**Pull request by samredai**
This adds the logic for the `Schema.find_column_name` method as well as adds a few tests.

### CODE REVIEW

File: python/src/iceberg/schema.py [hunk 1]
```diff
@@ -203,32 +210,32 @@ def after_map_value(self, value: NestedField) -> None:
...
     def map(self, map_type: MapType, key_result: T, value_result: T) -> T:
         """Visit a MapType"""
-        ...
+        ...  # pragma: no cover
 
     @abstractmethod
     def primitive(self, primitive: PrimitiveType) -> T:
         """Visit a PrimitiveType"""
-        ...
+        ...  # pragma: no cover
```
Comment (rdblue) [inline python/src/iceberg/schema.py:238]: What does this do?

File: python/src/iceberg/io/pyarrow.py [hunk 2]
```diff
@@ -42,12 +42,12 @@ class PyArrowFile(InputFile, OutputFile):
...
     Examples:
         >>> from iceberg.io.pyarrow import PyArrowFile
-        >>> input_file = PyArrowFile("s3://foo/bar.txt")
+        >>> # input_file = PyArrowFile("s3://foo/bar.txt")
         >>> # Read the contents of the PyArrowFile instance
         >>> # Make sure that you have permissions to read/write
         >>> # file_content = input_file.open().read()
 
-        >>> output_file = PyArrowFile("s3://baz/qux.txt")
+        >>> # output_file = PyArrowFile("s3://baz/qux.txt")
```
Comment (rdblue) [inline python/src/iceberg/io/pyarrow.py:50]: Is this change needed?

**Review by rdblue — APPROVED** [review]

File: python/src/iceberg/io/pyarrow.py (hunk 2 above)
Comment (samredai) [inline python/src/iceberg/io/pyarrow.py:50]: This change is unrelated to this PR but I noticed that since our tests run doc-strings now, this line and the line defining `input_file` was actually making a request to aws [here](https://github.com/apache/iceberg/blob/master/python/src/iceberg/io/pyarrow.py#L61) to determine the region. This just comments those lines out.

Comment (samredai) [reply python/src/iceberg/io/pyarrow.py:50]: I was dropping a comment [here](https://github.com/apache/iceberg/pull/4504/files#r843338128) just as you were commenting haha.

File: python/src/iceberg/schema.py (hunk 1 above)
Comment (samredai) [reply python/src/iceberg/schema.py:238]: This tells coverage to not include this line in the coverage % calculations.
